common/services/person/person.unformat.test.js

```diff
@@ -0,0 +1,271 @@
+            comments: 'explicitAssessmentValue',
+          },
+          {
+            key: 'ignored',
+            category: 'assessmentCategory',
+            assessment_question_id: 'ignoredId',
+            comments: 'ignoredValue',
+          },
+        ],
+      }
+    })
+
+    beforeEach(function() {
+      unformatted = unformat(person, fields, keys)
+    })
+
+    context('when asking for a property', function() {
+      before(function() {
+        fields = ['propertyField']
+      })
+      it('should return the value as is', function() {
+        expect(unformatted).to.deep.equal({ propertyField: 'propertyValue' })
+      })
+
+      context('but it has no value', function() {
+        before(function() {
+          fields = ['missingPropertyField']
+          keys = {
+            date: ['missingPropertyField'],
+          }
+        })
+        it('should return undefined', function() {
+          expect(unformatted).to.deep.equal({
+            missingPropertyField: undefined,
+          })
+        })
+      })
+    })
+
+    context('when asking for an identifier', function() {
+      before(function() {
+        fields = ['identifierField']
+        keys = {
+          identifier: ['identifierField'],
+        }
+      })
+      it('should return the identifier’s value', function() {
+        expect(unformatted).to.deep.equal({
+          identifierField: 'identifierValue',
+        })
+      })
+
+      context('but it has no value', function() {
+        before(function() {
+          fields = ['missingIdentifierField']
+          keys = {
+            date: ['missingIdentifierField'],
+          }
+        })
+        it('should return undefined', function() {
+          expect(unformatted).to.deep.equal({
+            missingIdentifierField: undefined,
+          })
+        })
+      })
+    })
+
+    context('when asking for a relationship', function() {
+      before(function() {
+        fields = ['relationshipField']
+        keys = {
+          relationship: ['relationshipField'],
+        }
+      })
+      it('should return the relationship’s id', function() {
+        expect(unformatted).to.deep.equal({
+          relationshipField: 'relationshipValue',
+        })
+      })
+
+      context('but it has no value', function() {
+        before(function() {
+          fields = ['missingRelationshipField']
+          keys = {
+            date: ['missingRelationshipField'],
+          }
+        })
+        it('should return undefined', function() {
+          expect(unformatted).to.deep.equal({
+            missingRelationshipField: undefined,
+          })
+        })
+      })
+    })
+
+    context('when asking for a date', function() {
+      before(function() {
+        fields = ['dateField']
+        keys = {
+          date: ['dateField'],
+        }
+      })
+      it('should return the value formatted as a date', function() {
+        expect(unformatted).to.deep.equal({ dateField: '1 Jan 2000' })
```

Comment:
I think it would be worth stubbing the date format method in this instance. 

If we change the default date format we don't want all these tests to break. We just need to know that it's calling the `formatDate` method with the correct value.

Reply:
Added